paasta_tools/firewall_logging.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import logging
+import os
+import signal
+import socket
+import sys
+
+import syslogmp
+from six.moves import socketserver
+
+from paasta_tools.firewall import services_running_here
+from paasta_tools.utils import _log
+from paasta_tools.utils import configure_log
+from paasta_tools.utils import load_system_paasta_config
+
+DEFAULT_NUM_WORKERS = 5
+
+log = logging.getLogger(__name__)
+
+
+class SyslogUDPHandler(socketserver.BaseRequestHandler):
+    def setup(self):
+        configure_log()
+        self.cluster = load_system_paasta_config().get_cluster()
+
+    def handle(self):
+        data, socket = self.request
+        syslog_to_paasta_log(data, self.cluster)
+
+
+def syslog_to_paasta_log(data, cluster):
+    iptables_log = parse_syslog(data)
+    if iptables_log is None:
+        return
+
+    service, instance = lookup_service_instance_by_ip(iptables_log['SRC'])
+    if service is None or instance is None:
+        return
+
+    # prepend hostname
+    log_line = iptables_log['hostname'] + ': ' + iptables_log['message']
+
+    _log(
+        service=service,
+        component='security',
+        level='debug',
+        cluster=cluster,
+        instance=instance,
+        line=log_line,
+    )
+
+
+def parse_syslog(data):
+    parsed_data = syslogmp.parse(data)
+    full_message = parsed_data.message.decode()
```

Comment:
we probably have quite a few syslog messages that don't decode as ascii or utf8 or whatever the locale will be, maybe decode as latin1 (so at least it never throws an exception) or just catch the exception and ignore the line (rather than crash the worker)?